### PR TITLE
Api startup

### PR DESCRIPTION
**Pull request by yPiuro**
#3

### CODE REVIEW

File: src/backend/mod.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+impl Api {
+    /// Index / Docs
+    #[oai(path = "/", method = "get", tag = ApiTags::API)]
+    async fn index(&self) -> Redirect {
+        Redirect::Response("/api/docs".to_string())
+    }
+}
+
+#[derive(poem_openapi::Tags)]
+enum ApiTags {
```
Comment (Notliam99) [inline src/backend/mod.rs:25]: Is this in the right order for a reading sense, should this be defined before the `Api` implementation

File: src/main.rs [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
-        StaticFilesEndpoint::new(env::current_dir().unwrap().join("notes_r_us_ui/dist"))
-            .index_file("index.html"),
-    );
+async fn main() -> std::result::Result<(), std::io::Error> {
+    let api_service = OpenApiService::new(
+        backend::Api,
+        "Notes R Us API Documentation",
+        std::env::var("CARGO_PACKAGE_VERSION").unwrap_or(String::from("N/A")),
+    )
+    .server("http://localhost:80")
```
Comment (Notliam99) [inline src/main.rs:13]: Should we add a Variable that holds the local server end point in one spot?

E.g.
```rust
let sever_str = format!("{}:{}", "0.0.0.0", 8000).as_str()
```

Also please pick a non http port for serving the server.

File: src/main.rs [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
-// use poem_openapi::{self, OpenApi}
 
 #[tokio::main]
-async fn main() {
-    let app = Route::new().nest(
-        "/",
-        StaticFilesEndpoint::new(env::current_dir().unwrap().join("notes_r_us_ui/dist"))
-            .index_file("index.html"),
-    );
+async fn main() -> std::result::Result<(), std::io::Error> {
```
Comment (Notliam99) [inline src/main.rs:7]: Optional I prefer unwraps to result enums, if you think otherwise, your call.

**Review by Notliam99 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (yPiuro) [reply src/main.rs:13]: Port 8000 is in use in my system, I will choose 3000 as that's what it was before, also for the server endpoint variable, we could have it be `localhost` if the program is compiled in `debug` mode and make it the actual URL if it's compiled in `release` mode.

What do you reckon?

Comment (yPiuro) [reply src/main.rs:13]: @Notliam99 what is going to be the URL

image

Comment (yPiuro) [reply src/main.rs:13]: Renamed var to `server_url` , as its only going to contain URL not port and https:// | http://.

**Comment by yPiuro** [comment]
Requested changes have almost fully been addressed, need review and some information before we are good to merge.

@Notliam99

File: src/main.rs (hunk 2 above)
Comment (Notliam99) [reply src/main.rs:13]: I think we'll make it notes.nzdev.org I'd have to move my notes from this subdomain tho.

Or something like notesrus.nzdev.org.

**Comment by yPiuro** [comment]
Ill make it notesrus.nzdev.org

Ill do it and commit when I get home, after that ill start working on the
routes.

**Comment by yPiuro** [comment]
@Notliam99 review and merge for me pls ❤️

**Review by Notliam99 — APPROVED** [review]
Wonder full looks Good.

Don't forget to add this too your notion.

**Pull request merged**